tests/pytorch/test_fused_optimizer.py

```diff
@@ -169,6 +171,87 @@ def test_frozen_model(self):
 
             torch.testing.assert_close(ref_param, tst_param)
 
+    def test_bf16_model_weight_cast(self):
+        dtype = torch.bfloat16
+        model = MultiheadAttention(
+            hidden_size=1024,
+            num_attention_heads=16,
+            layer_number=1,
+            params_dtype=dtype,
+            fuse_qkv_params=True,
+        ).cuda()
+        ref_params = []
+        master_params = []
+        model_params = []
+        for p in model.parameters():
+            if p.requires_grad:
+                ref_params.append(p.detach().clone().float())
+                master_params.append(p.detach().clone().float())
+                model_params.append(p)
+        options = {
+            "lr": 5e-4,
+            "betas": (0.9, 0.999),
+            "eps": 1e-08,
+            "weight_decay": 0,
+            "amsgrad": False,
+        }
+        ref_optim = torch.optim.Adam(ref_params, **options)
+        model_param_groups = [model_params]
+        tst_optim = te.optimizers.FusedAdam(
+            master_params, extra_param_groups=model_param_groups, fuse_dtype_casting=True, **options
+        )
+
+        for i in range(self.iters):
+            self.gen_grad(ref_params, master_params)
+            ref_optim.step()
+            tst_optim.step()
+            torch.testing.assert_close(ref_params, master_params)
+            model_params_to_fp32 = [p.float() for p in model_params]
+            torch.testing.assert_close(
+                ref_params, model_params_to_fp32, rtol=1e-3, atol=1e-3, equal_nan=True
+            )
+
+    def test_fp8_model_weight_cast(self):
```

Comment:
We should skip this test on systems that don't support FP8. See: https://github.com/NVIDIA/TransformerEngine/blob/c71e5e989ee09078249cfaf5ab27f24db059a329/tests/pytorch/test_float8tensor.py#L45

Reply:
Done